src/backend/lib/upgrade-storage.js

```diff
@@ -42,13 +43,44 @@ class Storage {
     log("..done. recreating temp directory..");
     mkdirp.sync(this.tmpdir);
     log("..done");
-
+    this.lock = new RWLock();
+    log("deleting old apks");
+    this.lock.writeLock(release => {
+      function done(err) {
+        if (err) throw new Error(err);
```

Comment:
What happens to the Node process (and/or Android app) if we throw an error & it doesn't get caught?

cc @gmaclennan